libraries/chain/include/eosio/chain/snapshot_detail.hpp

```diff
@@ -114,16 +114,13 @@ namespace eosio::chain::snapshot_detail {
       block_header                                        header;
       protocol_feature_activation_set_ptr                 activated_protocol_features;
       finality_core                                       core;
-      incremental_merkle_tree                             proposal_mtree;
-      incremental_merkle_tree                             finality_mtree;
       finalizer_policy_ptr                                active_finalizer_policy;
       proposer_policy_ptr                                 active_proposer_policy;
       flat_map<block_timestamp_type, proposer_policy_ptr> proposer_policies;
       flat_map<uint32_t, finalizer_policy_ptr>            finalizer_policies;
 
       // from block_state
-      using valid_t = uint32_t; // snapshot todo
-      std::optional<valid_t> valid;
+      std::optional<valid_t>                              valid;
```

Comment:
Yes, I remember you mentioned that it should always be present, and the std::optional was just because the `valid_t` is not be available when `block_state` is initially created. 

I assume that when the snapshot is initiated, and we save the chain head to the snapshot, we can `FC_ASSERT` that the `valid` member is populated, right?

Reply:
Yes, we can assert that the `valid` structure exists at the time the snapshot is being produced. It must always be the case.

In the snapshot itself, the `valid` structure must always be present, which is why I say it shouldn't be optional here. It should still be optional within `block_state`. But when loading the root `block_state` from a snapshot, `valid` should always end up populated.